Source/WebCore/svg/graphics/SVGImage.cpp

```diff
@@ -545,14 +569,32 @@ bool isInSVGImage(const Element* element)
     return page->chrome().client().isSVGImageChromeClient();
 }
 
-RefPtr<SVGImage> SVGImage::tryCreateFromData(std::span<const uint8_t> data)
+void SVGImage::subresourcesAreFinished(CompletionHandler<void()>&& completionHandler)
+{
+    ASSERT(rootElement());
+    Ref document = *internalPage()->localTopDocument();
+    if (document->loadEventFinished()) {
+        completionHandler();
+        return;
+    }
+
+    Ref window = *document->protectedWindow();
+    window->addEventListener(eventNames().loadEvent, SVGImageEventListener::create(window, WTFMove(completionHandler)), { /*capture */ false, /* passive */ std::nullopt, /* once */ true });
```

Comment:
It is unclear whether the completion handler will be called, say for instance if the window gets removed between the time of adding the event listener and the time the load event can actually fire.